src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -361,6 +395,25 @@ private IList<StringWithQualityHeaderValue> Sort(IList<StringWithQualityHeaderVa
             return sorted;
         }
 
+        private string GetFromCacheIfCompatible(string mediaType, Encoding encoding)
+        {
+            if (encoding != Encoding.UTF8)
+            {
+                return null;
+            }
+
+            foreach (var supportedMediaType in SupportedMediaTypes)
```

Comment:
Why is `GetMediaTypeWithCharset` looking at `SupportedMediaTypes`? We already know it's a 'valid' media-type, we just need to set the charset.